### ISSUE TITLE
Merge multiple notifications of a sync operation to a single notification

### ISSUE THREAD

**Issue by eranreshef-starkware** [argoproj/argo-cd#14849]
# Summary

It would be awesome if we (the users) will have the ability to combine multiple notification to a single notification in order to "reduce the noise" from ArgoCD"

# Motivation

When I run a sync operation for multiple apps at once, I get a separate notification per app for each phase of the sync process.
It would be nice if I could get 1 notification which contains a list of apps which were included in this sync execution and another notification which contains a list of sync results.

# Proposal
Basically, it means to add the option to have an operation-oriented notification mechanism along side with the existing application-oriented mechanism.
